internal/pkg/manifest/validate.go

```diff
@@ -734,3 +812,108 @@ func (d *DeadLetterQueue) Validate() error {
 func (*OverrideRule) Validate() error {
 	return nil
 }
+
+type validateDependenciesOpts struct {
+	mainContainerName string
+	sidecarConfig     map[string]*SidecarConfig
+	imageConfig       *Image
+}
+
+func validateDependencies(opts validateDependenciesOpts) error {
+	if err := validateImageDependsOnStatus(opts); err != nil {
+		return err
+	}
+	if err := validateSidecarsDependsOnStatus(opts); err != nil {
+		return err
+	}
+	return validateNoCircularDependencies(opts)
+}
+
+func validateImageDependsOnStatus(opts validateDependenciesOpts) error {
```

Comment:
```suggestion
func validateImageDependsOnEssentialContainerStatus(opts validateDependenciesOpts) error {
```

Another thought: since `validateImageDependsOnStatus` and `validateSidecarsDependsOnStatus` basically only validate whether an essential container dependency is specified correctly, what do you think of refactoring then into a function named
```go
validateEssentialContainerDependencies(dependsOn DependsOn)
```
and call by `validateEssentialContainerDependencies(i.imageConfig.DependsOn)` and `validateEssentialContainerDependencies(s.DependsOn)`